lib/internal/Magento/Framework/ObjectManager/Code/Generator/Repository.php

```diff
@@ -228,7 +228,6 @@ protected function _getGetMethod()
             'parameters' => [
                 [
                     'name' => 'id',
-                    'type' => 'int',
```

Comment:
As an example, you can write some repository interface with type hinting, specify preference via DI.xml to an auto-generated instance and run `setup:di:compile` command.